vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringBootAutoConfiguration.java

```diff
@@ -82,11 +82,16 @@ public ServletContextInitializer contextInitializer() {
                 new SpringServlet(context, rootMapping), mapping);
         registration.setInitParameters(initParameters);
         registration
-                .setAsyncSupported(configurationProperties.isAsyncSupported());
+                .setAsyncSupported(
+                        configurationProperties.getAsyncSupported() == null ?
```

Comment:
the configuration properties are all created as objects to know if the value comes from the application.properties or not (null values case). In this case the value is indeed initialized and that check should not be necessary but, as it can happen, I added the check. 

I could indeed check if there is a default value for the property and, if this happens, use the primitive instead of the object version, but I didn't want to write a more complex code for this. If you still think that something must be done, I could go for this

Reply:
>the configuration properties are all created as objects to know if the value comes from the application.properties or not

Does it matter whether the property is defined in the `application.properties` (which is in fact not the only way of setting the properties) or not ?

Previous  version of the class was also based on properties (let's say defined in `application.properties` ) but it had a default value `false` it the property is not available.

I understand that the code generation will be more complicated if we take into account boxing types (primitives and their boxed Java types).
And I'm kind if OK to avoid spending time on this considering that _most of_ properties are not used in fact via `VaadinConfigurationProperties`.

The only thing which I care:
- this PR changes _existing_ API. 
- previously there was `isAsyncSupported` with `boolean` as a property 
- previously there was `isLoadOnStartup` with `boolean` as a property 

Now the API is changed and the only reason why: the automatic class generation whose purpose is just provide IDE assistance.

I don't think it's a good thing to change API with a reason which has no relation to the API itself but rather just because of simplicity to generate the code.

So:
- I don't insist on _generic_ handling primitives in a proper way
- but API for `isLoadOnStartup` and `isAsyncSupported` should not be changed.

It's up to you how you want to deal with this: 
- consider these two properties as a special case and somehow "hardcode" them
- or fix them generally for any type.

Reply:
ok. The annotation processor has been generalized so `isLoadOnStartup` and `isAsyncSupported` remain